src/CMakeLists.txt

```
@@ -32,42 +32,38 @@ source_group("include_kea" FILES ${LIBKEA_H})
 add_library(${LIBKEA_LIB_NAME} ${LIBKEA_CPP} ${LIBKEA_H} )
 target_link_libraries(${LIBKEA_LIB_NAME} ${HDF5_LIBRARIES} ${HDF5_HL_LIBRARIES})
 
-if(BUILD_SHARED_LIBS)
-    SET_TARGET_PROPERTIES(${LIBKEA_LIB_NAME}
-            PROPERTIES
-            SOVERSION ${LIBKEA_VERSION_MAJOR}.${LIBKEA_VERSION_MINOR}
-            VERSION ${LIBKEA_VERSION}
-            CLEAN_DIRECT_OUTPUT 1)
-else()
-    SET_TARGET_PROPERTIES(${LIBKEA_LIB_NAME}
-            PROPERTIES
-            OUTPUT_NAME "${LIBKEA_LIB_NAME}"
-            PREFIX "lib"
-            VERSION ${LIBKEA_VERSION}
-            CLEAN_DIRECT_OUTPUT 1)
+include(GenerateExportHeader)
+generate_export_header(${LIBKEA_LIB_NAME}
+  EXPORT_MACRO_NAME KEA_EXPORT
+  EXPORT_FILE_NAME ${CMAKE_CURRENT_BINARY_DIR}/include/libkea/kea_export.h
+)
+target_include_directories(${LIBKEA_LIB_NAME} PUBLIC
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
+  $<INSTALL_INTERFACE:include>
+)
+
+set_target_properties(${LIBKEA_LIB_NAME}
+  PROPERTIES
+    CXX_VISIBILITY_PRESET hidden
+    VISIBILITY_INLINES_HIDDEN 1
+    PREFIX "lib"
+    SOVERSION ${LIBKEA_VERSION_MAJOR}.${LIBKEA_VERSION_MINOR}
+    VERSION ${LIBKEA_VERSION}
+)
+if(MSVC AND NOT BUILD_SHARED_LIBS)
+  set_target_properties(${LIBKEA_LIB_NAME} PROPERTIES OUTPUT_NAME "libkea")
 endif()
+
 add_library(Kealib INTERFACE)
 target_link_libraries(Kealib INTERFACE "${LIBKEA_LIB_NAME}")
```

This could be replaced by an alias target

```suggestion
add_library(Kealib::Kealib ALIAS ${LIBKEA_LIB_NAME})
```

and in CMakeLists of gdal driver, `${LIBKEA_GDAL_DRIVER}` would link to `Kealib::Kealib` instead of `Kealib`

---

What would be the benefit of doing this?
Would this mean that building the gdal driver in isolation would fail? (I do this in building the Arc driver for various versions - same kealib but different gdal versions)